README.md

```diff
@@ -26,75 +25,101 @@ $ vault secrets enable buddy
+    > **Warning**
+    > If no auto-rotation is set, the token should be generated with no expiration date.
 
-`insecure` - Disable SSL verification of API calls. You may need to set this to `true` if you are using Buddy On-Premises without signed certificate. Default: false
+- `token_ttl_in_days` – the lease time of the rotated root token in days. Default: `30`. Min: `2`
+- `base_url` – the Buddy API base URL. You may need to set this in your Buddy On-Premises API endpoint. Default: `https://api.buddy.works`
+- `insecure` – disables the SSL verification of the API calls. You may need to set this to `true` if you are using Buddy On-Premises without a signed certificate. Default: `false`
 
-## Rotate root token
+### Rotating root token
 
-Attempt to rotate the root credentials used to communicate with Buddy. Old token will be removed
+Updates the root credentials used for communication with Buddy. Rotating the root token removes the old one. To rotate the token, run `vault write -f buddy/rotate-root`.
 
 ```sh
 $ vault write -f buddy/rotate-root
 Success! Data written to: buddy/rotate-root
 ```
 
-## Roles
+## Vault token configuration
 
-Create a role and read its current credentials:
+### Creating role token
+
+To create a token with roles, run `vault write buddy/roles/ROLE_NAME` with the lease time and scopes.
+
+Example command for creating a token with the RUN_PIPELINE role:
 
 ```sh
-$ vault write buddy/roles/r1 \
+$ vault write buddy/roles/run_pipeline \
     ttl=30 \
     scopes=WORKSPACE,EXECUTION_RUN
-Success! Data written to: buddy/roles/r1    
+Success! Data written to: buddy/roles/run_pipeline   
 ```
 
-All options:
-
-`ttl` - Default lease for generated token. Vault will automatically revoke token after the duration. If not set or set to 0, will use system default.
-
-`max_ttl` - Maximum duration that generated token cab be extended to. If not set or set to 0, will use system default.
+Available options:
 
-`scopes` - The comma separated list of scopes
+- `ttl` – the default lease time for the generated vault token after which the token is automatically revoked. If not set or set to `0`, system default is used.
+- `max_ttl` – the maximum time the generated token can be extended to before it eventually expires. If not set or set to `0`, system default is used.
+- `scopes` – the [list of scopes](https://buddy.works/docs/api/getting-started/oauth2/introduction#supported-scopes) in the role, comma-separated.
+- `ip_restrictions` – the list of IP addresses to which the token is restricted, comma-separated.
+- `workspace_restrictions` – the list of workspace domains to which the token is restricted, comma-separated.
 
-`ip_restrictions` - The comma separated list of IP addresses
+### Reading role credentials
 
-`workspace_restrictions` - The comma separated list of workspace domains
-
-Read the credentials:
+To check the credentials in the role, run `read buddy/creds/ROLE_NAME`:
```

Comment:
w komendzie musi byc vault na poczatku bo to jest komenda a to co napisales to sa argumenty do niej czyli:
vault read buddy/creds/ROLE_NAME

Reply:
zreszta ponizej jest to powtorzone - nie wiem czy jest sens tak powtarzac mozna dac dwukropek i ponizej przyklad.

Reply:
zostawie tutaj to ROLE_NAME żeby było explicit że to jest przykładowe